### DISCUSSION TITLE
GLEDOPTO devices don't update in frontend

### DISCUSSION THREAD

**Discussion by hyperion007** [Koenkk/zigbee2mqtt#19379] [General]
Hi,

I have some devices that don't show correct information under "exposes" in the frontend. 
For example: 
Device: GL-C-006P 
It will show that it's OFF, I press "ON" in the frontend and the lights turn on but the frontend still shows "OFF". I have to press the refresh icon for the frontend to update to state "ON".
Same when I press "OFF", lights turn off but frontend still show "ON" until I hit the refresh icon.

I have QOS: 2 and retain: true

Both of my GL-C-006P devices behave this way.

Same issue with my GL-C-008P devices and GL-C-009P so it seems to be only with GLEDOPTO devices.

Zigbee2MQTT version
1.32.2 commit: 1ec1e57
Coordinator type
zStack3x0
Coordinator revision
20210120
Frontend version
0.6.133

**Comment by jmgutierrezm** [comment]
Hi, me too!
I have three GL-B-008P, the firmware update is detected but when I click on firmware update button, the update process is started but it never finish successfully.

This is the log:
Zigbee2MQTT:info  2023-10-31 12:25:52: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/update', payload '{"data":{"id":"0x680ae2fffed61e95"},"error":"Update of 'Bombilla Burbujita' failed (Timeout: device did not request any image blocks)","status":"error"}'
Zigbee2MQTT:error 2023-10-31 12:25:52: Update of 'Bombilla Burbujita' failed (Timeout: device did not request any image blocks)
Zigbee2MQTT:debug 2023-10-31 12:25:52: Error: Timeout: device did not request any image blocks
    at /app/node_modules/zigbee-herdsman-converters/src/lib/ota/common.ts:386:28
    at runNextTicks (node:internal/process/task_queues:61:5)
    at listOnTimeout (node:internal/timers:528:9)
    at processTimers (node:internal/timers:502:7)

Zigbee2MQTT version 1.33.1-1
Coordinator type Sonoff Zigbee 3.0 USB Dongle Plus

Thanks.

**Comment by hyperion007** [comment]
Are there any others here who use GLEDOPTO products that have the issue of the front end not showing the status of them?

**Comment by hyperion007** [comment]
This is becoming a bigger issue since any group that has a GLEDOPTO product will keep reporting "state: 'ON'" even though all devices in the group are off. All because of the GLEDOPTO state not refreshing automatically. As soon as I press the "refresh" on the ON/OFF for the device, the group publishes "state: 'OFF'"

**Comment by hyperion007** [comment]
I have now added reporting for OnOff, Level Control and Color Temperature for all but one of my GLEDOPTO devices. 
The [GL-C-006S](https://www.zigbee2mqtt.io/devices/GL-C-006S.html#gledopto-gl-c-006s) doesn't accept any reporting commands.

This works but it keeps flooding the network with publish data even without any changes to the status/levels. I will start to add filters for a lot of things attributes to get the network traffic down.

**Reply by OhmegaStar** [reply]
What have you set for min rep change, min rep interval, and max rep interval ?

**Comment by OhmegaStar** [comment]
I got the same issue with the Gledopto GL-SD-003P, got 5 of these installed, none updates state in z2m unless i refresh manually

**Comment by OhmegaStar** [comment]
I Added reporting on OnOff and LevelCtrl;
![image](https://github.com/Koenkk/zigbee2mqtt/assets/15914993/c7b3ed02-9543-4e92-b233-911b69c65965)

Now I get the missing updates as changes are made by physical contacts
Thanks for the clue on missing reporting

**Reply by Reinhard-M** [reply]
Thanks a lot for the hint, this was also my problem with a Gledopto GL-C-006P and a MiBox controller to switch light on/off and change brightness.

**Reply by ahuffman** [reply]
Thank you, used your suggestion with a GL-SD-301P and it worked like a champ.  Not sure why we have to do this for this device, but at least I was able to stumble upon this post to resolve it.

**Reply by rclarsfull** [reply]
Thx a lot. Worked for me too for GL-C-006P

**Comment by zengateway** [comment]
I have the same problem with Gledopto 5 in 1 Mini Led Controllers (GL-C-002P)
Sometimes devices won't refresh state until I manually press the refresh icon.
Zigbee2MQTT version   1.36.1
Coordinator type   zStack3x0
Coordinator revision   20230507
Frontend version     0.6.161
